From: Hannes Reinecke <hare@suse.de>
Date: Fri, 25 Aug 2017 15:22:22 +0200
Subject: [PATCH] lpfc: fixup crash during storage failover operations
Patch-Mainline: never, sle12 specific patch
References: bsc#1042847

During failover testing the lpfc driver would occasionally crash,
as the 'cmdiocbq' parameter in lpfc_sli4_els_wcqe_to_rspicbq()
might be NULL, but wasn't checked properly.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/lpfc/lpfc_sli.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_sli.c b/drivers/scsi/lpfc/lpfc_sli.c
index 5cf85fb..92ba3a5 100644
--- a/drivers/scsi/lpfc/lpfc_sli.c
+++ b/drivers/scsi/lpfc/lpfc_sli.c
@@ -12512,7 +12512,8 @@ lpfc_sli4_els_wcqe_to_rspiocbq(struct lpfc_hba *phba,
 	cmdiocbq = lpfc_sli_iocbq_lookup_by_tag(phba, pring,
 				bf_get(lpfc_wcqe_c_request_tag, wcqe));
 	/* Put the iocb back on the txcmplq */
-	lpfc_sli_ringtxcmpl_put(phba, pring, cmdiocbq);
+	if (cmdiocbq)
+		lpfc_sli_ringtxcmpl_put(phba, pring, cmdiocbq);
 	spin_unlock_irqrestore(&pring->ring_lock, iflags);
 
 	if (unlikely(!cmdiocbq)) {
-- 
1.8.5.6

